terraform/main.tf

```diff
@@ -1,6 +1,161 @@
-resource "azurerm_resource_group" "list" {
+resource "azurerm_resource_group" "rg" {
   for_each = var.resource_groups
 
   name     = each.key
-  location = each.value.location
+  location = each.value
 }
+
+
+resource "azurerm_virtual_network" "vnet" {
+
+  name                = var.vnets.name
+  address_space       = var.vnets.address_space
+  location            = var.vnets.location
+  resource_group_name = var.vnets.resource_group_name
+
+  depends_on = [azurerm_resource_group.rg]
+}
+
+
+
+resource "azurerm_subnet" "subnet" {
+  for_each = var.subnets
+
+  name                 = each.key
+  resource_group_name  = each.value.resource_group_name
+  virtual_network_name = each.value.virtual_network_name
+  address_prefixes     = each.value.address_prefixes
+  private_endpoint_network_policies_enabled = try(each.value.private_endpoint_network_policies_enabled, null)
+
+  depends_on = [
+    azurerm_resource_group.rg,
+    azurerm_virtual_network.vnet
+  ]
+}
+
+
+resource "azurerm_public_ip" "pip" {
+  name                = var.pip.name
+  location            = var.pip.location
+  resource_group_name = var.pip.resource_group_name
+  allocation_method   = var.pip.allocation_method
+  sku                 = var.pip.sku
+
+  depends_on = [
+    azurerm_resource_group.rg,
```

Comment:
trailing comma

Reply:
actually to chain dependencies, reference the rg resource object name for the value of the resource_group_name attr:

```
azurerm_resource_group.rg[var.pip.resource_group_name].Name
```